mozilla_version/test/test_fenix.py

```diff
@@ -0,0 +1,307 @@
+    ('32.0.1a2', PatternNotMatchedError),
+    ('32.0.1b2', PatternNotMatchedError),
+    ('32.0build0', ValueError),
+    ('32.0a1a2', PatternNotMatchedError),
+    ('32.0a1b2', PatternNotMatchedError),
+
+    ('55.0a2', PatternNotMatchedError),
+    ('56.0a2', PatternNotMatchedError),
+))
+def test_fenix_version_raises_when_invalid_version_is_given(version_string, ExpectedErrorType):
+    with pytest.raises(ExpectedErrorType):
+        FenixVersion.parse(version_string)
+
+
+@pytest.mark.parametrize('version_string, expected_type', VALID_VERSIONS.items())
+def test_fenix_version_is_of_a_defined_type(version_string, expected_type):
+    release = FenixVersion.parse(version_string)
+    assert getattr(release, 'is_{}'.format(expected_type))
+
+
+@pytest.mark.parametrize('previous, next', (
+    ('2.0.0', '3.0.0'),
+    ('2.0.0', '3.1.0'),
+    ('2.0.0', '3.0.1'),
+
+    ('2.0.1', '3.0.0'),
+    ('2.0.1', '2.1.0'),
+    ('2.0.1', '2.0.2'),
+
+    ('2.1.0', '3.0.0'),
+    ('2.1.0', '2.2.0'),
+    ('2.1.0', '2.1.1'),
+
+    ('2.0.0-beta.1', '3.0.0-beta.1'),
+    ('2.0.0-beta.1', '2.0.0-beta.2'),
+
+    ('2.0.0-beta.1', '2.0.0'),
+
+    ('1.0.0-rc.1', '1.0.0-rc.2'),
+    ('1.0.0-rc.1', '1.0.0'),
+    ('3.5.0-beta.4', '3.5.0-rc.2'),
+    ('3.5.0-beta.4', '3.5.0'),
+    ('3.5.0-rc.2', '3.5.0'),
+    ('3.5.0-rc.2', '3.5.0-rc.3'),
+))
+def test_fenix_version_implements_lt_operator(previous, next):
+    assert FenixVersion.parse(previous) < FenixVersion.parse(next)
+
+
+@pytest.mark.parametrize('equivalent_version_string', (
+    '3.00.0', '03.0.0', '3.0.0',
+))
+def test_fenix_version_implements_eq_operator(equivalent_version_string):
+    assert FenixVersion.parse('3.0.0') == FenixVersion.parse(equivalent_version_string)
+    # raw strings are also converted
+    assert FenixVersion.parse('3.0.0') == equivalent_version_string
+
+
+@pytest.mark.parametrize('equivalent_version_string', (
+    '3.00.0-beta.1', '03.0.0-beta.1', '3.0.0-beta.1',
+))
+def test_fenix_beta_version_implements_eq_operator(equivalent_version_string):
+    assert FenixVersion.parse('3.0.0-beta.1') == FenixVersion.parse(equivalent_version_string)
+    # raw strings are also converted
+    assert FenixVersion.parse('3.0.0-beta.1') == equivalent_version_string
+
+
+@pytest.mark.parametrize('equivalent_version_string', (
+    '3.00.0-rc.1', '03.0.0-rc.1', '3.0.0-rc.01',
+))
+def test_fenix_rc_version_implements_eq_operator(equivalent_version_string):
+    assert FenixVersion.parse('3.0.0-rc.1') == FenixVersion.parse(equivalent_version_string)
+    # raw strings are also converted
+    assert FenixVersion.parse('3.0.0-rc.1') == equivalent_version_string
+
+
+@pytest.mark.parametrize('wrong_type', (
+    3,
+    3.0,
+    ('3', '0', '1'),
+    ['3', '0', '1'],
+    LooseVersion('3.0'),
+    StrictVersion('3.0'),
+))
+def test_fenix_version_raises_eq_operator(wrong_type):
+    with pytest.raises(ValueError):
+        assert FenixVersion.parse('3.0') == wrong_type
```

Comment:
Nit: `s/3.0/3.0.0` to get to 100% coverage https://coveralls.io/builds/27610891/source?filename=mozilla_version%2Ffenix.py#L98